packages/nc-gui-v2/components/smartsheet-toolbar/ShareView.vue

```diff
@@ -1,13 +1,165 @@
 <script lang="ts" setup>
-import MdiOpenInNew from '~icons/mdi/open-in-new'
-import { useUIPermission } from '#imports'
+import { useClipboard } from '@vueuse/core'
+import { ViewTypes } from 'nocodb-sdk'
+import { computed } from 'vue'
+import { useToast } from 'vue-toastification'
+import { useNuxtApp } from '#app'
+import { useSmartsheetStoreOrThrow } from '~/composables/useSmartsheetStore'
+import { extractSdkResponseErrorMsg } from '~/utils/errorUtils'
+import MdiOpenInNewIcon from '~icons/mdi/open-in-new'
+import MdiCopyIcon from '~icons/mdi/content-copy'
 
 const { isUIAllowed } = useUIPermission()
+const { view, $api } = useSmartsheetStoreOrThrow()
+
+const { copy } = useClipboard()
+const toast = useToast()
+const { $e } = useNuxtApp()
+const { dashboardUrl } = useDashboard()
+
+let showShareModel = $ref(false)
+const passwordProtected = $ref(false)
+const shared = ref()
+
+const allowCSVDownload = computed({
+  get() {
+    return !!(shared.value?.meta && typeof shared.value.meta === 'string' ? JSON.parse(shared.value.meta) : shared.value.meta)
+      ?.allowCSVDownload
+  },
+  set(allow) {
+    shared.value.meta = { allowCSVDownload: allow }
+    saveAllowCSVDownload()
+  },
+})
+
+const genShareLink = async () => {
+  shared.value = await $api.dbViewShare.create(view.value.id as string)
+  // shared.meta = shared.meta && typeof shared.meta === 'string' ? JSON.parse(shared.meta) : shared.meta;
+  // // todo: url
+  // shareLink = shared;
+  // passwordProtect = shared.password !== null;
+  // allowCSVDownload = shared.meta.allowCSVDownload;
+  showShareModel = true
+}
+
+const sharedViewUrl = computed(() => {
+  if (!shared.value) return
+  let viewType
+
+  switch (shared.value.type) {
+    case ViewTypes.FORM:
+      viewType = 'form'
+      break
+    case ViewTypes.KANBAN:
+      viewType = 'kanban'
+      break
+    default:
+      viewType = 'view'
+  }
+
+  // todo: get dashboard url
+  return `${dashboardUrl?.value}/nc/${viewType}/${shared.value.uuid}`
+})
+
+async function saveAllowCSVDownload() {
+  try {
+    const meta = shared.value.meta && typeof shared.value.meta === 'string' ? JSON.parse(shared.value.meta) : shared.value.meta
+
+    // todo: update swagger
+    await $api.dbViewShare.update(shared.value.id, {
+      meta,
+    } as any)
+    toast.success('Successfully updated')
+  } catch (e) {
+    toast.error(await extractSdkResponseErrorMsg(e))
+  }
+  if (allowCSVDownload?.value) {
+    $e('a:view:share:enable-csv-download')
+  } else {
+    $e('a:view:share:disable-csv-download')
+  }
+}
+
+const saveShareLinkPassword = async () => {
+  try {
+    await $api.dbViewShare.update(shared.value.id, {
+      password: shared.value.password,
+    })
+    toast.success('Successfully updated')
+  } catch (e) {
+    toast.error(await extractSdkResponseErrorMsg(e))
+  }
+
+  $e('a:view:share:enable-pwd')
+}
+
+const copyLink = () => {
+  copy(sharedViewUrl?.value as string)
+  toast.info('Copied to clipboard')
```

Comment:
maybe use a [`message`](https://antdv.com/components/message) for this? @dstala What do you say?
Toast seems strange for the clipboard message

Reply:
> 

what would be the difference between these two? if `message`, where will it be displayed?

Reply:
Message looks neat, will update with that